File: src/main/java/gestorAplicacion/ventas/Producto.java
package gestorAplicacion.ventas;

import java.io.Serial;
import java.io.Serializable;
import java.util.List;

/**
 * Clase que representa un producto dentro del sistema para la venta.
 *
 * @author Juan Daniel Bula, Leidy Daniela Alzate, Oscar Andres Usuga
 */
public class Producto implements Serializable {

    @Serial
    private static final long serialVersionUID = 3L;
    private static List<Producto> productosVendidos;

    private String codigo;
    private String nombre;
    private double precio;
    private LineaProducto linea;
    private int cantidad;

    public Producto(String codigo, String nombre, double precio, int cantidad) {
        this.codigo = codigo;
        this.nombre = nombre;
        this.precio = precio;
        this.cantidad = cantidad;
    }

    public Producto(String codigo, String nombre, double precio, LineaProducto linea, int cantidad) {
        this(codigo, nombre, precio, cantidad);
        this.linea = linea;
    }

    public String getCodigo() {
        return codigo;
    }
}

File: python/gestorAplicacion/ventas/producto.py
class Producto:
    productos_vendidos = []

    def __init__(self, codigo, nombre, precio, cantidad):
        self._codigo = codigo
        self._nombre = nombre
        self._precio = precio
        self._cantidad = cantidad

    def getCodigo(self):
        return self._codigo

    def getNombre(self):
        return self._nombre

    def getCantidad(self):
        return self._cantidad

    def setCantidad(self, cantidad):
        self._cantidad = cantidad

    def venderProducto(self):
        Producto.productos_vendidos.append(self)
        self._cantidad -= 1

    def __str__(self):
        return "Producto [codigo = %s, nombre = %s, precio = %f]" % (self._codigo, self._nombre, self._precio)
